examples/chess_position.py

```diff
@@ -0,0 +1,159 @@
+                                [-7.1482e+00, -1.1688e+01, -6.5647e+00,  1.1144e+01, -9.7372e+00, -1.1413e+01, -1.0610e+01, -1.6807e+01, -1.3919e+01, -5.2683e+00, -1.4328e+01, -1.5948e+01],
+                                [-3.4779e+00, -6.1333e+00, -1.9338e+00,  7.6642e+00, -5.6737e+00, -4.8410e+00, -8.8620e+00, -1.1995e+01, -8.9921e+00, -4.4965e+00, -1.1039e+01, -1.0286e+01],
+                                [-5.4471e+00, -7.3961e+00, -4.0091e+00,  1.0008e+01, -8.2364e+00, -7.7552e+00, -1.1562e+01, -1.4871e+01, -1.2803e+01, -4.9956e+00, -1.4799e+01, -1.4024e+01],
+                                [-1.0975e+01, -1.4695e+01, -7.0197e+00,  1.6814e+01, -1.4073e+01, -1.4442e+01, -1.8432e+01, -2.4743e+01, -2.0148e+01, -6.8818e+00, -2.2039e+01, -2.4336e+01],
+                                [-1.8632e+00,  1.4883e+00, -2.7764e+00, -1.0044e+00,  1.1251e+01, -2.1246e+00, -9.2420e+00, -1.0971e+01, -8.4337e+00, -4.7349e+00, -1.9218e+00, -9.4412e+00],
+                                [-5.1858e+00,  1.4478e+01, -6.0291e+00, -2.7641e+00, -3.3264e+00, -3.5799e+00, -1.1613e+01,  6.5402e-01, -1.1856e+01, -6.3279e+00, -1.2178e+01, -1.0819e+01],
+                                [-8.3045e+00, -1.1004e+01, -5.4839e+00,  1.2478e+01, -1.1314e+01, -1.0851e+01, -1.3917e+01, -1.9016e+01, -1.5132e+01, -3.3467e+00, -1.7661e+01, -1.8405e+01],
+                                [-1.0578e+01, -9.6113e+00, -1.0476e+01, -6.9775e+00, -3.4868e+00, -9.2788e+00,  2.2917e-01,  1.6189e+00, -1.9423e+00,  9.8658e-01,  8.7442e+00,  1.9664e+00],
+                                [ 1.7821e+00,  2.5520e+00,  2.3929e+00, -3.7030e-01, -3.3090e-01,  1.9798e-01, -6.8749e+00, -6.4092e+00, -5.7582e+00, -5.6637e+00, -8.7920e+00, -7.5211e+00]])
+
+occupancy_in_board = np.array([False, False, False, False, False, False,  True, False,
+                                True, False, False, False, False, False, False, False, 
+                               False,  True, False, False,  True,  True, False, False, 
+                               False,  True,  True, False,  True, False,  True,  True,
+                                True, False,  True,  True,  True, False, False, False,
+                               False, False, False,  True, False,  True, False,  True,
+                               False, False,  True, False,  True, False,  True, False,
+                                True, False, False, False,  True, False, False, False])
+
+# Variable to represent the whole board
+board = intvar(0, 12, (8,8), "board")
+
+# Variable to represent the pieces, with first row the probabilies and second row the indexes
+pieces = intvar(0, 10000000, (2, len(piece_probabilities)), name="pieces")
+
+# Make the probabilites into reasonable big values
+pieces_values = np.empty([len(piece_probabilities), len(piece_probabilities[0])], dtype=int)
+for i in range(len(piece_probabilities)):
+    for j in range(len(piece_probabilities[0])):
+        pieces_values[i,j] = int((piece_probabilities[i,j]*10000).item())
```

Comment:
Basically you convert the "probabilities" (not really probabilities) in big values. Why not just divide with a small number defining the precision? e.g.

PRECISION = 1e-4
pieces_values = piece_probabilities / PRECISION
pieces_values = cpm_array(np.int64(lprobs)) # This is to convert to int + to convert the array to cpm_array, so that you can directly use it in the objective function